peps/pep-0750.rst

```diff
@@ -53,837 +33,1089 @@ deferred pending further experience with f-strings. Work on this PEP was
+authors who provide interesting machinery built with template strings.
 
 All three groups can start from an important framing:
 
-- Existing solutions (such as template engines) can do parts of tag strings
-- But tag strings move logic closer to "normal Python"
+- Existing solutions (such as template engines) can do parts of template strings
+- But template strings move logic closer to "normal Python"
 
-Consumers can look at tag strings as starting from f-strings:
+Consumers can look at template strings as starting from f-strings:
 
 - They look familiar
 - Scoping and syntax rules are the same
 
-They first thing they need to absorb: unlike f-strings, the string isn't
-immediately evaluated "in-place". Something else (the tag function) happens.
-That's the second thing to teach: the tag functions do something particular.
-Thus the concept of "domain specific languages" (DSLs). What's extra to
-teach: you need to import the tag function before tagging a string.
+The first thing they need to absorb: unlike f-strings, template string literals
+don't evaluate to strings; rather, they evaluate to a new type, ``Template``.
+``Template`` is a simple type intended to be used by template processing code.
+It's not until you call that code that you get the result you want: typically,
+a string, although processing code can of course return any arbitrary type.
 
-Tag function authors think in terms of making a DSL. They have
-business policies they want to provide in a Python-familiar way. With tag
+Template function authors think in terms of making a DSL. They have
+business policies they want to provide in a Python-familiar way. With template
 functions, Python is going to do much of the pre-processing. This lowers
 the bar for making a DSL.
 
-Tag authors can begin with simple use cases. After authors gain experience, tag strings can be used to add larger
-patterns: lazy evaluation, intermediate representations, registries, and more.
+Template authors can begin with simple use cases. After authors gain experience, 
+template strings can be used to add larger patterns, like parsing to intermediate
+representations or context-sensitive processing of interpolations.
 
 Each of these points also match the teaching of decorators. In that case,
 a learner consumes something which applies to the code just after it. They
 don't need to know too much about decorator theory to take advantage of the
 utility.
 
-Common Patterns Seen In Writing Tag Functions
-=============================================
+
+Common Patterns Seen in Processing Templates
+============================================
 
 Structural Pattern Matching
 ---------------------------
 
-Iterating over the arguments with structural pattern matching is the expected
-best practice for many tag function implementations:
+Iterating over the ``Template.args`` with structural pattern matching is the expected
+best practice for many template function implementations:
```

Comment:
Can you expand on why this is seen as better than e.g. `isinstance` checks?

Reply:
The expectation (admittedly unproven so far) is that developers will often want to write interesting pattern matches against `Interpolation`.